modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1463,408 +1416,51 @@ private void dropTableLocally(long causalityToken, CatalogTableDescriptor tableD
-                .thenCompose(tableIds -> inBusyLock(busyLock, () -> {
-                    var tableFuts = new CompletableFuture[tableIds.size()];
-
-                    var i = 0;
+    private CompletableFuture<List<Table>> tablesAsyncInternalBusy() {
+        HybridTimestamp now = clock.now();
 
-                    for (int tblId : tableIds) {
-                        tableFuts[i++] = tableAsyncInternal(tblId, false);
-                    }
-
-                    return allOf(tableFuts).thenApply(unused -> inBusyLock(busyLock, () -> {
-                        var tables = new ArrayList<Table>(tableIds.size());
-
-                        for (var fut : tableFuts) {
-                            var table = fut.join();
-
-                            if (table != null) {
-                                tables.add((Table) table);
-                            }
-                        }
+        return anyOf(schemaSyncService.waitForMetadataCompleteness(now), stopManagerFuture)
+                .thenComposeAsync(unused -> inBusyLockAsync(busyLock, () -> {
+                    int catalogVersion = catalogService.activeCatalogVersion(now.longValue());
 
-                        return tables;
-                    }));
-                }));
-    }
+                    Collection<CatalogTableDescriptor> tableDescriptors = catalogService.tables(catalogVersion);
 
-    /**
-     * Collects a list of direct table ids.
-     *
-     * @return A list of direct table ids.
-     */
-    private List<Integer> directTableIds() {
-        return configuredTablesCache.configuredTableIds();
-    }
+                    if (tableDescriptors.isEmpty()) {
+                        return completedFuture(List.of());
+                    }
 
-    /**
-     * Gets direct id of table with {@code tblName}.
-     *
-     * @param tblName Name of the table.
-     * @return Direct id of the table, or {@code null} if the table with the {@code tblName} has not been found.
-     */
-    @Nullable
-    private Integer directTableId(String tblName) {
-        try {
-            TableConfiguration exTblCfg = directProxy(tablesCfg.tables()).get(tblName);
+                    CompletableFuture<TableImpl>[] tableImplFutures = tableDescriptors.stream()
+                            .map(tableDescriptor -> tableAsyncInternalBusy(tableDescriptor.id()))
+                            .toArray(CompletableFuture[]::new);
 
-            if (exTblCfg == null) {
-                return null;
-            } else {
-                return exTblCfg.id().value();
-            }
-        } catch (NoSuchElementException e) {
-            return null;
-        }
+                    return allOf(tableImplFutures)
+                            .thenApply(unused1 -> inBusyLock(
+                                    busyLock,
+                                    () -> Stream.of(tableImplFutures).map(CompletableFuture::join).collect(toList())
```

Comment:
BusyLock is used to forbid accessing internals, when it is unsafe, e.g. on node stop.
Here, collecting tables doesn't need to be in busy lock.

Reply:
fix it